### PR TITLE
PLT-1680 Converting client to react-router. Stage 1

### PR DESCRIPTION
**Pull request by crspeller**

### CODE REVIEW

**Comment by lfbrock** [comment]
Here's a couple things I saw (testing on Safari):

1. Missing "Add" button beside username in "Add Members to Channel" modal (not sure if this was from this or from Harrison's changes)

screenshot 2016-03-04 10 23 40

2. Not able to edit/delete my own message (doesn't show up as an option)
screenshot 2016-03-04 10 26 39

3. Clicking on a permalink adds a blank space to the bottom of the window
screenshot 2016-03-04 10 27 08

4. Discard changes modal after not saving a theme color change is missing button text
screenshot 2016-03-04 10 29 12

**Comment by crspeller** [comment]
Thanks! @lfbrock 

Can you confirm that you are not seeing these problems on bladekick as well?

**Comment by lfbrock** [comment]
Looks like the Discard Changes modal button is on CI, and I figured out the "Remove Member" button in 1) doesn't show up  for your own account so it's actually not a bug. 

All the other ones are not on CI.

**Comment by crspeller** [comment]
Thanks! Added to known issues.

File: api/context.go [hunk 1]
```diff
@@ -105,6 +86,11 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	c.RequestId = model.NewId()
 	c.IpAddress = GetIpAddress(r)
 
+	/*	if !h.isApi {
```
Comment (crspeller) [inline api/context.go]: Remove this

File: api/oauth.go [hunk 2]
```diff
@@ -20,6 +23,11 @@ func InitOAuth(r *mux.Router) {
 
 	sr.Handle("/register", ApiUserRequired(registerOAuthApp)).Methods("POST")
 	sr.Handle("/allow", ApiUserRequired(allowOAuth)).Methods("GET")
+	sr.Handle("/{service:[A-Za-z]+}/complete", AppHandlerIndependent(completeOAuth)).Methods("GET")
```
Comment (crspeller) [inline api/oauth.go]: Support old routes.

File: api/team.go [hunk 3]
```diff
@@ -391,9 +390,6 @@ func isTeamCreationAllowed(c *Context, email string) bool {
 }
 
 func getAll(c *Context, w http.ResponseWriter, r *http.Request) {
```
Comment (crspeller) [inline api/team.go]: Check if team directory enabled.

File: mattermost.go [hunk 4]
```diff
@@ -53,6 +53,7 @@ func main() {
 
 	utils.InitTranslations()
 	utils.LoadConfig(flagConfigFile)
+	utils.InitHTML()
```
Comment (crspeller) [inline mattermost.go]: Remove.

File: model/utils.go [hunk 5]
```diff
@@ -112,6 +112,14 @@ func GetMillis() int64 {
 	return time.Now().UnixNano() / int64(time.Millisecond)
 }
 
+func BoolToJson(boolean bool) string {
```
Comment (crspeller) [inline model/utils.go]: Remove.

File: web/react/components/center_panel.jsx [hunk 6]
```diff
@@ -25,40 +25,43 @@ export default class CenterPanel extends React.Component {
...
-    onUserChange() {
-        this.setState({user: UserStore.getCurrentUser(), profiles: JSON.parse(JSON.stringify(UserStore.getProfiles()))});
+    componentWillUnmount() {
+        PreferenceStore.removeChangeListener(this.onStoresChange);
+        ChannelStore.removeChangeListener(this.onStoresChange);
+        UserStore.removeChangeListener(this.onStoresChange);
     }
     render() {
-        const channel = ChannelStore.getCurrent();
+        if (!this.validState) {
```
Comment (crspeller) [inline web/react/components/center_panel.jsx]: this is a function.

File: web/react/components/post_body.jsx [hunk 7]
```diff
@@ -80,12 +80,10 @@ class PostBody extends React.Component {
                     username = parentPost.props.override_username;
                 }
 
-                if (global.window.mm_locale === 'en') {
-                    if (username.slice(-1) === 's') {
-                        apostrophe = '\'';
-                    } else {
-                        apostrophe = '\'s';
-                    }
+                if (username.slice(-1) === 's') {
```
Comment (crspeller) [inline web/react/components/post_body.jsx]: Check locale store.

File: web/react/components/signup_team.jsx [hunk 8]
```diff
@@ -47,9 +47,9 @@ export default class TeamSignUp extends React.Component {
     }
 
     render() {
-        var teamListing = null;
+        let teamListing = null;
 
-        if (global.window.mm_config.EnableTeamListing === 'true') {
+        /*if (global.window.mm_config.EnableTeamListing === 'true') {
```
Comment (crspeller) [inline web/react/components/signup_team.jsx]: Fix.

File: web/react/pages/root.jsx [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+
+    GlobalActions.emitPostFocusEvent(postId);
+}
+
+function onLoggedOut(nextState) {
+    const teamName = nextState.params.team;
+    Client.logout(
+        () => {
+            browserHistory.push('/' + teamName + '/login');
+            BrowserStore.signalLogout();
```
Comment (crspeller) [inline web/react/pages/root.jsx]: Take a look at this.

File: web/react/stores/channel_store.jsx [hunk 10]
```diff
@@ -67,6 +68,15 @@ class ChannelStoreClass extends EventEmitter {
     removeChangeListener(callback) {
         this.removeListener(CHANGE_EVENT, callback);
     }
+    emitSelectedChange() {
```
Comment (crspeller) [inline web/react/stores/channel_store.jsx]: Remove this?

**Pull request merged**